### ISSUE TITLE
Full-screen mode

### ISSUE THREAD

**Issue by mikeny07** [pygfx/wgpu-py#451]
Is it possible to run pygfx applications in a full-screen mode without the top "title" bar?

**Comment by Korijn** [comment]
No. We'll have to add support for it here in the `wgpu.gui` subpackage.

**Comment by almarklein** [comment]
Actually ... this works:

```
import PySide6
from wgpu.gui.qt import WgpuCanvas, run
import pygfx as gfx

canvas = WgpuCanvas(size=(900, 400))
canvas.showFullScreen()
renderer = gfx.renderers.WgpuRenderer(canvas)

... etc
```

And for glfw:
```
canvas = WgpuCanvas(size=(900, 400))

import glfw

glfw.set_window_monitor(canvas._window, glfw.get_primary_monitor(), 0, 0, 1920, 1080, glfw.DONT_CARE)
...
```

Would be nice to support `WgpuCanvas(... fullscreen=True)`, but until then, you can use something like the above.

**Comment by Korijn** [comment]
> Would be nice to support `WgpuCanvas(... fullscreen=True)`, but until then, you can use something like the above.

Let's use this issue to track the API changes we would propose.

* `WgpuCanvas(..., fullscreen=True)`
* `WgpuCanvas().fullscreen(windowless=True/False)` to switch at runtime
* `WgpuCanvas().windowed()` to switch at runtime

**Comment by almarklein** [comment]
> I guess it would also require some method to get a list of monitors?

I think it can default to the default / current screen. Qt's `showFullScreen` has no args to specify a monitor.

This might be a good time to also consider other [window states](https://doc.qt.io/qt-6/qt.html#WindowState-enum: normal (windowed), hidden, minimized, maximixed, fullscreen.

**Comment by mikeny07** [comment]
Great, the `glfw.set_window_monitor ` function works. 

It would be nice to allow the user to set the full-screen mode.

Thank you so much for your excellent work!

**Comment by hmaarrfk** [comment]
Are you really trying to abstract away Qt and GLFW. Why not just document your short examples. 

I strongly doubt any code beyond simple examples can be made to run agnosticqlly between the two
